File: Projects/Project_Cinema Website/my-react-app/src/services/SeatService.js
import axios from "axios";

const baseURL = 'http://localhost:8080/api/seats'; // Replace with your actual API endpoint

class SeatService {

    // GET
    static async getSeatById(seatId) {
        return (await axios.get(`${baseURL}/getSeat/${seatId}`)).data;
    }

    static async getAllSeats() {
        return (await axios.get(`${baseURL}/getAllSeats`)).data;
    }

    static async getSeatsByHallId(hallId) {
        return (await axios.get(`${baseURL}/getSeatsByHallId/${hallId}`)).data;
    }

    // POST
    static async addSeat(newSeat) {
        return (await axios.post(`${baseURL}/addSeat`, newSeat)).data;
    }

    // PUT (Update)
    static async updateSeatById(seatId, updatedSeat) {
        return (await axios.put(`${baseURL}/updateSeat/${seatId}`, updatedSeat)).data;
    }

    // DELETE
    static async deleteSeatById(seatId) {
        return (await axios.delete(`${baseURL}/deleteSeat/${seatId}`)).data;
    }
}

export default SeatService;

File: Projects/Project_Cinema Website/backend/src/main/java/com/cinema_website/backend/service/SeatService.java
package com.cinema_website.backend.service;

import com.cinema_website.backend.dto.SeatDTO;
import com.cinema_website.backend.repository.SeatRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class SeatService {

    private final SeatRepository seatRepository;

    @Autowired
    public SeatService(SeatRepository seatRepository) {
        this.seatRepository = seatRepository;
    }

    // GET
    public SeatDTO getSeat(int seatId) {
        return seatRepository.getSeatById(seatId);
    }

    public List<SeatDTO> getAllSeats() {
        return seatRepository.getAllSeats();
    }

    public List<SeatDTO> getSeatsByHallId(int hallId) {
        return seatRepository.getSeatsByHallId(hallId);
    }

    // INSERT
    public boolean addSeat(SeatDTO seat) {
        return seatRepository.addSeat(seat);
    }

    // UPDATE
    public boolean updateSeat(int seatId, SeatDTO updatedSeat) {
        return seatRepository.updateSeatById(seatId, updatedSeat);
    }

    // DELETE
    public boolean deleteSeat(int seatId) {
        return seatRepository.deleteSeatById(seatId);
    }
}
